File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleEntityDto.cs
using System;
using Volo.Abp.ObjectExtending;

namespace Volo.Abp.Application.Dtos;

[Serializable]
public abstract class ExtensibleEntityDto<TKey> : ExtensibleObject, IEntityDto<TKey>
{
    /// <summary>
    /// Id of the entity.
    /// </summary>
    public TKey Id;// { get; set; }

    protected ExtensibleEntityDto()
        : this(true)
    {

    }

    protected ExtensibleEntityDto(bool setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }

    @Override public string toString()
    {
        return $"[DTO: {GetType().Name}] Id = {Id}";
    }
}

[Serializable]
public abstract class ExtensibleEntityDto : ExtensibleObject, IEntityDto
{
    protected ExtensibleEntityDto()
        : this(true)
    {

    }

    protected ExtensibleEntityDto(bool setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }

    @Override public string toString()
    {
        return $"[DTO: {GetType().Name}]";
    }
}

File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleEntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import com.ueueo.data.objectextending.ExtensibleObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public abstract class ExtensibleEntityDto extends ExtensibleObject implements IEntityDto {
    /**
     * Id of the entity.
     */
    private ID id;

    protected ExtensibleEntityDto() {
        this(true);

    }

    protected ExtensibleEntityDto(boolean setDefaultsForExtraProperties) {
        super(setDefaultsForExtraProperties);
    }

    @Override
    public String toString() {
        return String.format("[DTO: %s] Id = %s", getClass().getName(), id.toString());
    }
}
